### PR TITLE
Bfw 863 wait dialog refactor

### PR DESCRIPTION
**Pull request by michalrudolf**

### CODE REVIEW

File: src/gui/dialogs/window_dlg_wait.c [hunk 1]
```diff
@@ -68,32 +63,35 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
...
+            render_icon_align(icon_rc, IDR_PNG_wizard_icon_hourglass, COLOR_BLACK, ALIGN_CENTER);
 
-        if (window->flags & DLG_W8_HOUR_REDRW) {
             const uint16_t x = icon_rc.x;
             const uint16_t y = icon_rc.y;
 
-            render_icon_align(icon_rc, IDR_PNG_wizard_icon_hourglass, COLOR_BLACK, ALIGN_CENTER);
-            if (window->flags & 0x0010) {
+            // if (window->animation == ANIM_START) -> draw only icon
+            if (window->animation == ANIM_SAND_1) {
```
Comment (DRracer) [inline src/gui/dialogs/window_dlg_wait.c:80]: how about a switch(){...}?

Comment (DRracer) [reply src/gui/dialogs/window_dlg_wait.c:80]: And the sequences of lines can be also reduced by using some arrays of coordinates to draw. Not a blocker though, just says something about the author's programming abilities ;)

File: src/gui/dialogs/window_dlg_wait.c [hunk 2]
```diff
@@ -105,39 +103,57 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
...
-        rc_pro.h = 30;
-        snprintf(text, text_len, "%d%%", window->progress);
-        render_text_align(rc_pro, text, window->font_title, window->color_back, window->color_text, window->padding, ALIGN_CENTER);
-        window->flags &= 0x7FFF;
+    if (window->components & DLG_W8_DRAW_PROGRESS) { // progress bar enabled
+        if (window->progress_chng) {                 // progress changed
+            window->progress_chng = false;
+
+            char text[8];
+            rect_ui16_t rc_pro = rc;
```
Comment (DRracer) [inline src/gui/dialogs/window_dlg_wait.c:114]: what is this assignment for?

File: src/gui/dialogs/window_dlg_wait.c [hunk 3]
```diff
@@ -105,39 +103,57 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
...
+            rc_pro.y += 120;
+            display->fill_rect(rc_pro, COLOR_GRAY);
+
+            if (window->progress != -1) {
+                const uint16_t w = rc_pro.w;
+                rc_pro.w = (w * window->progress) / 100;
+                display->fill_rect(rc_pro, COLOR_ORANGE);
+                rc_pro.x += rc_pro.w;
+                rc_pro.w = w - rc_pro.w;
+                sprintf(text, "%d%%", window->progress);
```
Comment (DRracer) [inline src/gui/dialogs/window_dlg_wait.c:127]: sNprint!!!!

File: src/gui/dialogs/window_dlg_wait.c [hunk 4]
```diff
@@ -105,39 +103,57 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
...
+            } else {
+                strcpy(text, "N/A");
```
Comment (DRracer) [inline src/gui/dialogs/window_dlg_wait.c:129]: strlcpy!!!! how many times shall I say that?

Comment (michalrudolf) [reply src/gui/dialogs/window_dlg_wait.c]: commit `0ace17e`

File: src/gui/dialogs/window_dlg_wait.c (hunk 3 above)
Comment (michalrudolf) [reply src/gui/dialogs/window_dlg_wait.c]: commit `0ace17e`

File: src/gui/dialogs/window_dlg_wait.c (hunk 2 above)
Comment (michalrudolf) [reply src/gui/dialogs/window_dlg_wait.c]: Not all members, but here you go:
commit `0ace17e`

File: src/gui/dialogs/window_dlg_wait.c (hunk 1 above)
Comment (michalrudolf) [reply src/gui/dialogs/window_dlg_wait.c]: This hardcoded coordinates are so ugly it is not in my powers to make it better. I tried, but it's probably less readable..
commit `0ace17e`

**Pull request merged**